### PR TITLE
Refactor: Better config processing

### LINKED ISSUE #14: Better config parser procedure
As title mentioned, the parser of config file should be reworked.

### PR DESCRIPTION
**Pull request by h-alice**
Now the config processor doesn't do the image processing stuff.

### CODE REVIEW

**Comment by NatLee** [comment]
天啊 重構 uwu

**Comment by h-alice** [comment]
@NatLee 我要讓他變真正能用的工具uwu

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 *.exe
-vendor
\ No newline at end of file
+vendor
+.DS_Store
```
Comment (NatLee) [inline .gitignore]: Where is `desktop.ini`? 🥲🥲🥲

File: main.go [hunk 2]
```diff
@@ -110,7 +110,7 @@ func main() {
 		}
 	}
 
-	conf, err := config.LoadConfig(config_path)
+	conf, err := config.LoadConfigFromFile(config_path)
```
Comment (NatLee) [inline main.go:74]: Naming is good. 🥳

File: config/config_processor.go [hunk 3]
```diff
@@ -12,15 +13,71 @@ import (
...
+// Factor: Resize factor.
+//
+// Algorithm: Resize algorithm. Either `nearestneighbor`, `catmullrom`, or `approxbilinear`.
+//
+// NOTE: The `Factor` is prioritized over `Width` and `Height`.
+type ResizeConfig struct {
+	Width     int     `yaml:"width"`     // Output image width
+	Height    int     `yaml:"height"`    // Output image height
+	Factor    float32 `yaml:"factor"`    // Resize factor
+	Algorithm string  `yaml:"algorithm"` // Resize algorithm
```
Comment (NatLee) [inline config/config_processor.go:44]: Could you tell me if there's a definition for the algorithm?

File: config/config_processor.go [hunk 4]
```diff
@@ -12,15 +13,71 @@ import (
...
+
+// Currently not used.
+type OutputDirConfig struct {
+	DirName *string `yaml:"dirName"` // Output directory name
+}
+
+// Config structure for config file.
+//
+// Profiles: List of profile configurations.
+type ConfigFileRoot struct {
```
Comment (NatLee) [inline config/config_processor.go:71]: I feel that `Root` is not an appropriate name. 🤔

**Review by NatLee — APPROVED** [review]
These commits are excellent and are ready to be merged in.

File: config/config_processor.go (hunk 3 above)
Comment (h-alice) [reply config/config_processor.go:44]: Yes, there are 3 kinds of interpolation algorithm implemented in GoLang.
- `catmullrom`: Provides best quality.
- `Bilinear`: Familiar name, provides balanced quality and speed.
- `nearest neighborhood`: Your art will become pixel art.

File: config/config_processor.go (hunk 4 above)
Comment (h-alice) [reply config/config_processor.go:71]: I think so, any better idea?
This is for the root-node of parsed config.

File: main.go (hunk 2 above)
Comment (h-alice) [reply main.go:74]: Long name is better than ambiguous name. 🎉

File: .gitignore (hunk 1 above)
Comment (h-alice) [reply .gitignore]: It will be ignored in next config.
Also, who uses windows for developing? : D

**Pull request merged**